model/controller.go

```diff
@@ -34,6 +36,9 @@ type Controller interface {
 	// for a service.
 	AppendInstanceHandler(f func(*ServiceInstance, Event)) error
 
+	// AppendIngressHandler notifies about changes to ingress resources
+	AppendIngressHandler(f func(*v1beta1.Ingress, Event)) error
+
```

Comment:
The ingress config seems to be a subset of the model's [`RouteRule`](https://github.com/istio/manager/blob/master/model/proxy/alphav1/config/cfg.pb.go#L153) (that's what you mean by `ProxyConfig`, right)? So, we can have the controller convert ingresses to such rules, and have the proxy watcher subscribe to these.